src/remote_worker.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (C) 2018 dcurl Developers.
+ * Use of this source code is governed by MIT license that can be
+ * found in the LICENSE file.
+ */
+
+#include "constants.h"
+#include "dcurl.h"
+#include "remote_common.h"
+
+#define CHANNEL 1
+
+int main(int argc, char const *const *argv)
+{
+    char trytes[TRANSACTION_TRYTES_LENGTH];
+    char buf[4];
+    int mwm;
+
+    amqp_connection_state_t conn;
+    amqp_envelope_t envelope;
+
+    dcurl_init();
+
+    if (!connect_broker(&conn, CHANNEL))
+        goto fail_to_exit;
+
+    if (!set_consuming_queue(&conn, CHANNEL, "incoming_queue"))
+        goto fail_to_exit;
+
+    for (;;) {
+        if (!consume_message(&conn, CHANNEL, &envelope))
+            goto fail_to_exit;
+
+        printf(
```

Comment:
Don't invoke `printf` directly. Instead, ensure conditional build and optional diagnosis mode for message display.